testng-core/src/main/java/org/testng/TestRunner.java

```diff
@@ -558,6 +556,13 @@ private void initMethods() {
             comparator);
   }
 
+  private static ITestNGMethod[] classLevelConfigs(ITestClass tc, boolean beforeConfig) {
```

Comment:
As it is only used once for each, I don't see the value add of this method. I think you can inline the line where they are called.

Reply:
Its extracted out as a method, because if I inline this, then the code becomes unreadable.

So `fixMethodsWithClass(classLevelConfigs(tc, true), tc, beforeClassMethods);` will now become

`fixMethodsWithClass(ITestClassConfigInfo.allBeforeClassMethods(tc).toArray(ITestNGMethod[]::new), tc, beforeClassMethods);`

which is a bit too verbose.

Reply:
Fixed